trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/signing/Signer.java

```diff
@@ -154,10 +136,30 @@ static String sign(
                 ? legacyAws4Signer.sign(requestToSign, signingParams)
                 : aws4Signer.sign(requestToSign, signingParams);
 
-        return signedRequest.firstMatchingHeader("Authorization").orElseThrow(() -> {
+        String authorization = signedRequest.firstMatchingHeader("Authorization").orElseThrow(() -> {
             log.debug("Signer did not generate \"Authorization\" header");
             return new WebApplicationException(Response.Status.BAD_REQUEST);
         });
+
+        return buildInternalSigningContext(authorization, signingParams, xAmzDate);
+    }
+
+    private static InternalSigningContext buildInternalSigningContext(String authorization, AwsS3V4SignerParams signingParams, String xAmzDate)
+    {
+        Map<String, String> parts = Splitter.on(',').trimResults().withKeyValueSeparator('=').split(authorization);
```

Comment:
Could some of this logic be rolled into `ParsedAuthorization`?

Reply:
Yeah, good call. I wrote this before that existed.